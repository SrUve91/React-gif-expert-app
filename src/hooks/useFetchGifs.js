import { useEffect, useState } from "react"
import { getGifts } from "../helpers/getGifs";

export const useFecthGifs = (category) =>{

    const [state, setstate] = useState({
        data:[],
        loading:true
    });

    useEffect(()=>{

        getGifts(category).then(imgs =>{
            setstate({
                data: imgs,
                loading:false
            });
        });

    },[category]);

    return state; //{data:[], loading: true};
}