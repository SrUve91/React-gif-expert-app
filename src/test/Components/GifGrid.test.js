import React from 'react'
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFecthGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {

    const category = "One Punch";
    test('Se debe mostrar correctamente', () => {
    useFecthGifs.mockReturnValue({
        data:[],
        loading:true
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se carga imagenes con useFetchGifs', () => {

        const gifs=[{
            id:'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id:'DFG',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFecthGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    });
});
