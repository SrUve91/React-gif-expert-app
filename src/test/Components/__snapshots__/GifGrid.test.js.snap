// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Se debe mostrar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid debe de mostrar items cuando se carga imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
    <GifGridItem
      id="DFG"
      key="DFG"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
